builder/test/copy_pipeline.cc

```diff
@@ -176,13 +182,181 @@ TEST_P(copied_result, pipeline) {
           correct += in_buf(x + dx, y + dy);
         }
       }
-      ASSERT_EQ(correct, out_buf(x, y)) << x << " " << y;
+      ASSERT_EQ(correct, out_buf(x - offset_x, y - offset_y));
+    }
+  }
+
+  ASSERT_EQ(eval_ctx.heap.total_count, 0);
+}
+
+TEST_P(copied_result, two_results) {
+  int schedule = std::get<0>(GetParam());
+  int offset_x = std::get<1>(GetParam());
+  int offset_y = std::get<2>(GetParam());
+
+  // Make the pipeline
+  node_context ctx;
+
+  auto in = buffer_expr::make(ctx, "in", 2, sizeof(short));
+  auto out1 = buffer_expr::make(ctx, "out1", 2, sizeof(short));
+  auto out2 = buffer_expr::make(ctx, "out2", 2, sizeof(short));
+
+  auto intm = buffer_expr::make(ctx, "intm", 2, sizeof(short));
+
+  var x(ctx, "x");
+  var y(ctx, "y");
+
+  // In this pipeline, the result is copied to the output. We should just compute the result directly in the output.
```

Comment:
I'm confused, how does the aliasing work if there are two output buffers?

Reply:
The pipeline is basically:
```
f: in -> intm
copy1: intm -> out1
copy2: intm -> out2
```
After aliasing, it looks like:
```
f: in -> out1
copy2: out1 -> out2
```
It works because aliasing rewrites intm to be make_buffer of out1, which gets used for both f's output and copy2's input, and because both outputs consume the same region of intm.

That said, it's missing a lot of checks right now. They're missing on main too, so this branch isn't a regression, just exposing a lot of holes in aliasing that I'm working on.